include/util/AppendVariant.h

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) TrajoptLib contributors
+
+#pragma once
+
+#include <variant>
+
+// NOLINTBEGIN
+template <typename... Vs, typename... Aps>
+std::variant<Vs..., Aps...> _append_variant(std::variant<Vs...> myvar,
+                                            Aps... newvals) {
+  return std::variant<Vs..., Aps...>();
```

Comment:
This returns a default-constructed variant instead of `myvar` or `newvals`.

Reply:
If you look at the way it's used, the function body doesn't matter. It's used in a `decltype()` block which iirc doesn't get value-evaluated. The point of the function isn't to take two variants and create a new one with the combined contents, it just gets the type that contains the types of the variant and other types listed.

Reply:
The behavior and intended usage should be documented.

Reply:
Ok, on it